### PR TITLE
Make all the things

### PR DESCRIPTION
**Pull request by martinpitt**
See individual commits for details. I tested all of make-{dist,srpm,rpm,deb} from a completely clean git checkout, with downloading and with self-building.

### CODE REVIEW

**Comment by martinpitt** [comment]
@marusak: This should also fix your make-rpms woes.

File: tools/make-debs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+rm -rf "$builddir"
+mkdir -p "$builddir"
+olddir=$(pwd)
+cd "$builddir"
+tar --strip-components=1 -xf "$source"
+cp -r tools/debian .
+# plug in our version
+sed -i "1 s/(0-/(${version}-/" debian/changelog
+dpkg-buildpackage -us -uc -b -j$(nproc)
+cd "$olddir"
```
Comment (KKoukiou) [inline tools/make-debs:74]: `cd -` also works here, just to avoid storing olddir.

Comment (martinpitt) [reply tools/make-debs:74]: Done, thanks!

**Comment by martinpitt** [comment]
I fixed up the messaging harder, and added another fix for a bug that @marusak just found - we shouldn't download a tarball if there are local uncommitted changes.

**Comment by marusak** [comment]
Nice, thank you for fixing up both issues I've struggled with today morning!

**Review by KKoukiou — APPROVED** [review]
I have already tests the make-debs in my livetest branch btw, works as expected.

**Pull request merged**